Replace usage of ``transformClient`` (to be removed in the next release of YARP)

The ``frameTransformClient`` device is going to be removed in the next version of YARP and thus we need to update the code using it.
See https://github.com/robotology/yarp/pull/2999

Related issues: 
- https://github.com/ami-iit/yarp-device-openxrheadset/issues/40
- https://github.com/robotology/walking-teleoperation/issues/131

cc @lrapetti

With @davidegorbani we tried today but we had issues. In particular, we had this error continuosly:
```
[ERROR] |yarp.device.FrameTransformClient| getTransform(): Frames  root_link_desired  and  openxr_head  are not connected
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[ERROR] |yarp.device.FrameTransformClient| getTransform(): Frames  root_link_desired  and  openxr_head  are not connected
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_right_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[ERROR] |yarp.device.FrameTransformClient| getTransform(): Frames  root_link_desired  and  openxr_head  are not connected
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[ERROR] |yarp.device.FrameTransformClient| getTransform(): Frames  root_link_desired  and  openxr_head  are not connected
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::openxr_head
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_right_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::vive_tracker_right_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::vive_tracker_right_elbow_pose
[WARNING] HumanStateProvider : Failed to read link orientation from virtual link sensor  TransformServer::pose::vive_tracker_left_elbow_pose
[WARNING] HumanStateProvider : Failed to read link position from virtual link sensor  TransformServer::pose::openxr_head
```
The error is not appearing constantly, but a couple of times per second. The visualization has some jumps when this happens. 
We have the transform server running on a Windows machine, while the client on an Ubuntu machine on the same network.
I have tried to edit the timeout (specified in https://github.com/robotology/yarp/blob/65a3a508c0b5147bdf5106ac173370382486004e/src/libYARP_dev/src/yarp/dev/FrameTransformContainer.h#L99 and not set from conf file) from 0.2s to 1.0s on both machines, but nothing changed. I wonder if this problem has been fixed in recent version of ``yarp``. Currently we are using 3.7.2,  although I don't see anything particular in the changelog

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+- Update iframetransform_to_iwear.xml to use new TF device (https://github.com/robotology/wearables/pull/202)
+
 ### Added
 - CMake: Permit to explictly specify Python installation directory by setting the `WEARABLES_PYTHON_INSTALL_DIR` CMake variable (https://github.com/robotology/wearables/pull/197).
 

--- a/devices/IFrameTransformToIWear/conf/iframetransform_to_iwear.xml
+++ b/devices/IFrameTransformToIWear/conf/iframetransform_to_iwear.xml
@@ -2,10 +2,11 @@
 <!DOCTYPE robot PUBLIC "-//YARP//DTD yarprobotinterface 3.0//EN" "http://www.yarp.it/DTD/yarprobotinterfaceV3.0.dtd">
 <robot name="IFrameTransformToIWear" build=0 portprefix="">
 
-<device type="transformClient" name="TransformClient">
+<device type="frameTransformClient" name="TransformClient">
     <param name="period">0.01</param>
-    <param name="local">/tf</param>
-    <param name="remote">/transformServer</param>
+    <param name="filexml_option">ftc_yarp_only.xml</param>
+    <param name="ft_client_prefix">/IFrameTransformToIWear/tf</param>
+    <param name="local_rpc">/IFrameTransformToIWear/tf/local_rpc</param>
 </device>
 
 <device type="iframetransform_to_iwear" name="IFrameTransformToIWear">

